p4-16/spec/grammar.mdk

```diff
@@ -27,47 +31,60 @@ nonTypeName
     | ENTRIES
     | TYPE
     ;
+// END:nonTypeName
 
+// BEGIN:name
 name
     : nonTypeName
     | LIST
     | TYPE_IDENTIFIER
     ;
+// END:name
 
+// BEGIN:nonTableKwName
 nonTableKwName
    : IDENTIFIER
    | TYPE_IDENTIFIER
    | APPLY
    | STATE
    | TYPE
    ;
+// END:nonTableKwName
 
 optCONST
     : /* empty */
     | CONST
     ;
 
+// BEGIN:optAnnotations
 optAnnotations
     : /* empty */
     | annotations
     ;
+// END:optAnnotations
 
+// BEGIN:annotations
 annotations
     : annotation
     | annotations annotation
     ;
+// END:annotations
 
+// BEGIN:annotation
 annotation
     : "@" name
     | "@" name "(" annotationBody ")"
     | "@" name "[" structuredAnnotationBody "]"
     ;
+// END:annotation
 
+// BEGIN:annotationBody
 annotationBody
     : /* empty */
     | annotationBody "(" annotationBody ")"
     | annotationBody annotationToken
     ;
+// END:annotationBody
```

Comment:
how come annotationToken below doesn't need a comment?

Reply:
Given the process that I used to make these changes, if a rule for a nonterminal does no have BEGIN/END comments around it, it is because it was never used in an INCLUDE statement added to the file P4-16-spec.mdk.  Why not?  Two possible reasons that would not be a mistake on my part: (a) the nonterminal is not defined in P4-16-spec.mdk anywhere, or (b) the nonterminal is defined in P4-16-spec.mdk, but in the places where it is, its right-hand side is not identical to what is in grammar.mdk, and I did not want to decide in this PR which of those should be considered correct.

Reply:
I just double-checked: there is no definition for `annotationToken` in the P4-16-spec.mdk file that I started making the changes in this PR.  There IS a mention of `annotationToken` in the text before `annotationBody`'s definition is shown, but it appears `annotationToken` either never made it into the grammar definition in the body of the spec there, or someone deleted it at some point in the past.